crates/pixi_utils/src/reqwest.rs

```diff
@@ -95,7 +95,10 @@ pub fn oci_middleware() -> OciMiddleware {
     OciMiddleware
 }
 
-pub fn build_reqwest_clients(config: Option<&Config>) -> (Client, ClientWithMiddleware) {
+pub fn build_reqwest_clients(
+    config: Option<&Config>,
+    s3_config_project: Option<HashMap<String, rattler_networking::s3_middleware::S3Config>>,
```

Comment:
we could pass an empty hashmap instead of `None` here as well but i thought it's a bit clearer in the API when we actually are using project config and when not like this